### PR TITLE
Feature/issue 1350 topics scroll offset

### PR DESCRIPTION
**Pull request by tomy-pg**

### CODE REVIEW

File: src/components/topics/task_list_component.tsx [hunk 1]
```diff
@@ -70,14 +70,24 @@ export class TaskListComponent extends React.PureComponent<Props, State> {
     }
 
     componentDidUpdate(previousProps: Props): void {
-        if (this.flatListRef && hasContentChanged(previousProps, this.props)) {
+        console.log(this.props.scrollOffset)
+        if (this.flatListRef && hasTopicDetailPageChanged(previousProps, this.props)) {
```
Comment (rasmus-storjohann-PG) [inline src/components/topics/task_list_component.tsx:73]: Legibility will be improved if you do an early return on `this.flatListRef` being null.

How about if you had a local variable like so
```ts
const foo = {
                ...this.state,
                data: this.props.tasks,
                scrollOffset: this.state.scrollOffset,
            };
```
and then you could have this in one call
```ts
            return this.setState(
                            foo, 
                            (): void => scrollToOffsetWithTimeout(this.flatListRef, this.state.scrollOffset),
                    );
```
and this in the other
```ts
            return this.setState(
                     { ...foo, scrollOffset: 0}, 
                    (): void => scrollToOffsetWithTimeout(this.flatListRef, this.state.scrollOffset),
            );
```

File: src/components/topics/task_list_component.tsx [hunk 2]
```diff
@@ -135,9 +145,13 @@ const scrollToOffsetWithTimeout = (flatListRef: FlatListRef, offset: number): vo
...
-const hasContentChanged = (previousProps: Props, props: Props): boolean => {
-    return previousProps.headerContentIdentifier !== props.headerContentIdentifier || tasksHaveChanged(previousProps, props);
-};
+const hasTopicDetailPageChanged = (previousProps: Props, props: Props): boolean => (
+    previousProps.headerContentIdentifier !== props.headerContentIdentifier
+);
+
+const hasTopicListDataChanged = (previousProps: Props, props: Props): boolean => (
+    tasksHaveChanged(previousProps, props)
+)
```
Comment (rasmus-storjohann-PG) [inline src/components/topics/task_list_component.tsx]: Do you need this function? it's effectively just an alias.

**Review by rasmus-storjohann-PG — CHANGES_REQUESTED** [review]

File: src/components/topics/task_list_component.tsx (hunk 2 above)
Comment (tomy-pg) [reply src/components/topics/task_list_component.tsx]: Makes sense

File: src/components/topics/task_list_component.tsx (hunk 1 above)
Comment (tomy-pg) [reply src/components/topics/task_list_component.tsx:73]: The earlier return for `this.flatListRef` makes a lot of sense. 

As for the local variable, the values inputted into this `setState` changes in the last commit to clean it up a bit.

**Review by rasmus-storjohann-PG — APPROVED** [review]
I like it!

**Pull request merged**